### PR TITLE
Added custom generator features

### PR DESCRIPTION
**Pull request by dennisroche**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/yeoman/environment/blob/main/lib/generator-features.js and commit https://github.com/yeoman/environment/commit/02b5d27cd57c62a94fc75aa82abd6a9939c68f22
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by dennisroche** [comment]
Reverted `prettier` changes as too much churn and would be difficult to review

File: types/yeoman-environment/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for yeoman-environment 2.10
+// Type definitions for yeoman-environment 3.4.1
```
Comment (manuth) [inline types/yeoman-environment/index.d.ts:1]: Please revert this change - I don't think it's necessary as you didn't edit the `yeoman-environment`-typings

File: types/yeoman-generator/index.d.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for yeoman-generator 5.0
+// Type definitions for yeoman-generator 5.3.0
```
Comment (manuth) [inline types/yeoman-generator/index.d.ts:1]: Please remove the trailing `.0` as specifying patch-versions isn't allowed and thus cause the tests to fail.

You might want to write `5.2` here as this was (as far as I can see) the first version that introduced features in that manner.

File: types/yeoman-environment/index.d.ts (hunk 1 above)
Comment (manuth) [reply types/yeoman-environment/index.d.ts:1]: If you, for some reason, would like to bump the version of `@types/yeoman-environment` you have to change the version-number to `3.4`, but I don't think it's necessary, tho

File: types/yeoman-generator/index.d.ts [hunk 3]
```diff
@@ -227,6 +227,21 @@ declare namespace Generator {
         path: string;
     }
 
+    /**
+     * Represents generators feature
+     */
+    interface GeneratorFeatures {
```
Comment (manuth) [inline types/yeoman-generator/index.d.ts:233]: You might want to add all the other features (`uniqueBy`, `unique`, `tasksMatchingPriority`) the user could provide:
https://github.com/yeoman/generator/blob/c1c847dbed005e40ded29ea46d1efd4dbac5cc51/lib/index.js#L122-L130

**Review by neurolag — CHANGES_REQUESTED** [review]
Thanks a ton for your contribution!
I have a small number of changes that I'd recommend you to consider applying

File: types/yeoman-generator/index.d.ts (hunk 2 above)
Comment (dennisroche) [reply types/yeoman-generator/index.d.ts:1]: Done

**Review by neurolag — APPROVED** [review]
Awesome!
Looks good to me 😊

**Comment by dennisroche** [comment]
Ready to merge

**Comment by dennisroche** [comment]
Ready to merge
